src/main/java/com/aws/emr/profiler/core/requestprocessor/HTTPRequestProcessor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file.
+ * This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aws.emr.profiler.core.requestprocessor;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class HTTPRequestProcessor {
+    private static OkHttpClient httpClient = new OkHttpClient();
+    private static String INSTANCE_METADATA_ENDPOINT = "169.254.169.254";
+    private static String INSTANCE_METADATA_URI = "http://%s/latest/meta-data/local-hostname";
+
+    public HTTPRequestProcessor() {
+        this.INSTANCE_METADATA_URI = String.format(INSTANCE_METADATA_URI, INSTANCE_METADATA_ENDPOINT);
+    }
+
+    public HTTPRequestProcessor(String host) {
+        this.INSTANCE_METADATA_URI = String.format(INSTANCE_METADATA_URI, host);
```

Comment:
Is this to support external metadata servers?

Reply:
No, ec2 supports querying this endpoint for the instance metadata. check out this link for more info: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

Reply:
I meant is the second constructor there to support non 169.254.169.254 ips?